### ISSUE TITLE
Implement CI-based distribution

### ISSUE THREAD

**Issue by ronaldtse** [riboseinc/riffol#16]
We'd like to cross-compile the Riffol binary on different systems. Rust standard library depends on `libc`, which can be different on different Linux distros (Debian vs CentOS).

This task is to create a Jenkinsfile that can distribute standalone binaries that target different platforms.

Links:

* https://grahamenos.com/rust-osx-linux-musl.html
* https://os.phil-opp.com/freestanding-rust-binary/
* https://www.reddit.com/r/rust/comments/2geibf/how_to_distribute_programs_written_in_rust_as/

The resulting binaries should be distributed in the Releases section in this repo.

**Comment by ronaldtse** [comment]
@drystone for testing, we could use Travis and its matrices. 

My intention for Jenkins was for the distribution of releases, especially on our own yum repo at https://github.com/riboseinc/yum

**Comment by drystone** [comment]
@ronaldtse, I guess we are going to need both in that case. Matrices would indeed be useful for testing and they appear broken in Jenkins - at least with a declarative Jenkinsfile.

Jenkins is a bit pointless at the moment as all it does is test. I suggest we leave it in as a basis for building packages and binary distribution - perhaps not add the GitHub hooks for the time being.

We could, however, hook both this and the Nereond repo to Travis so we can do the cross platform testing. I'll push some .travis.yml in the meantime.

**Comment by ronaldtse** [comment]
@drystone you're absolutely right. Let's use Travis to test. I think it's easy to package Riffol / Nereond in RPMs using Jenkins as long as the build flow allows it.
